crux/src/Crux/SVCOMP.hs

```diff
@@ -31,44 +35,94 @@ import Crux.Config
 import Crux.Config.Common
 
 
-data SVCOMPOptions = SVCOMPOptions
+data SVCOMPOptions (f :: Type -> Type) = SVCOMPOptions
   { svcompBlacklist :: [FilePath]
       -- ^ Benchmarks to skip when evaluating in SV-COMP mode
 
-  , svcompMemlimit :: Maybe Integer
-      -- ^ total memory usage limit (in MB) per verification task
+  , svcompArch       :: f Arch
+      -- ^ The architecture assumed for the verification task
 
-  , svcompCPUlimit :: Maybe Integer
-      -- ^ CPU time limit (in seconds) per verification task
+  , svcompSpec       :: f FilePath
+      -- ^ The file containing the specification text used to verify the program. Likely a .prp file.
 
-  , svcompOutputFile :: FilePath
-      -- ^ file path for JSON verification results output
+  , svcompWitnessDir :: FilePath
+      -- ^ The directory in which to output the witness automaton file
   }
 
-svcompOptions :: Config SVCOMPOptions
+data Arch = Arch32 | Arch64
+  deriving (Show,Eq,Ord)
+
+archSpec :: ValueSpec Arch
+archSpec =
+  (Arch32 <$ atomSpec "32bit") <!>
+  (Arch64 <$ atomSpec "64bit")
+
+parseArch :: (Arch -> opts -> opts) -> String -> OptSetter opts
+parseArch mk = \txt opts ->
+  case txt of
+    "32bit" -> Right (mk Arch32 opts)
+    "64bit" -> Right (mk Arch64 opts)
+    _       -> Left "Invalid architecture"
+
+-- | We cannot verify an SV-COMP program without knowing the architecture and
+-- the specification. Unfortunately, Crux's command-line argument parser makes
+-- it somewhat awkward to require certain arguments (without defaults). As a
+-- hacky workaround, we parse the command-line arguments for the
+-- architecture/specification as if they were optional and later check here to
+-- see if they were actually provided, throwing an error if they were not
+-- provided.
+processSVCOMPOptions :: SVCOMPOptions Maybe -> IO (SVCOMPOptions Identity)
```

Comment:
This is an extremely gross hack I came up with to make certain command-line arguments required. I hate it. If there is a better way to do this in Crux, please let me know.